src/providers/caching/route/model/cached-route.ts

```diff
@@ -44,24 +45,21 @@ export class CachedRoute<Route extends SupportedRoutes> {
     return this.route.protocol;
   }
 
-  // TODO: ROUTE-217 - Support native currency routing in V4
-  public get tokenIn(): Token {
-    return this.route.input.wrapped;
+  public get currencyIn(): Currency {
+    return this.route.input;
```

Comment:
this affected v3 and v4 caching though right? for v3 we want the same and for v4 we want something different right?

Reply:
I think it's ok. The reason is because type Currency is a union type of Token and NativeCurrency, which means it's backward compatible with v2/v3/mixed cached routes input and output type of Token, meanwhile v4 cached routes can have either a Token or NativeCurrency token.